lms/static/js/utils/dropdown.js

```diff
@@ -0,0 +1,217 @@
+                menu: $('.dropdown-menu'),
+                menu_inactive: 'is-hidden',
+                menu_active: 'is-visible',
+                menu_align: 'align-',
+                text_align: 'control-text-align'
+            },
+
+            init: function(parent) {
+
+                if (_dropdown.opts.button.length && _dropdown.opts.menu.length) {
+
+                    if (parent) {
+                        _dropdown.opts.page = $(parent);
+                    }
+
+                    _dropdown.listenForClick();
+                    _dropdown.listenForKeypress();
+                }
+            },
+
+            listenForClick: function() {
+
+                _dropdown.opts.button.on('click', function() {
+                    _dropdown.closeDropdownMenu(); // close any open menus
+                    _dropdown.openDropdownMenu($(this)); // then open the chosen menu
+                });
+
+                _dropdown.opts.page.on('click', function() {
+                    _dropdown.closeDropdownMenu();
+                });
+            },
+
+            listenForKeypress: function() {
+
+                _dropdown.opts.page.on('keydown', function(e) {
+                    var keyCode = e.keyCode,
+                        focused = $(e.currentTarget.activeElement),
+                        items, menu;
+
+                    if (27 === keyCode) {
+                        // When the ESC key is pressed, close all menus
+                        _dropdown.closeDropdownMenus();
+                    }
+
+                    if (focused.is('.action')) {
+                        // Key handlers for when a menu item has focus
+                        menu = focused.closest('.dropdown-menu');
+
+                        switch (keyCode) {
+                            case 38: // // UP
+                                _dropdown.previousMenuItemLink(focused, menu);
+                                break;
+
+                            case 40: // DOWN
+                                _dropdown.nextMenuItemLink(focused, menu);
+                                break;
+
+                            case 27: // ESC
+                                _dropdown.closeDropdownMenus();
+                                break;
+                        }
+
+                    } else if (focused.is('.has-dropdown')) {
+                        // Key handlers for when the button that opens the menu has focus
+                        switch(keyCode) {
+                            case 40: // DOWN
+                                _dropdown.openDropdownMenu(focused);
+                                break;
+
+                            case 13: // ENTER
+                                _dropdown.openDropdownMenu(focused);
+                                break;
+
+                            case 27: // ESC
+                                _dropdown.closeDropdownMenus();
+                                break;
+                        }
```

Comment:
It appears there are only 2 cases here if (keyCode === 40 || keyCode === 13) else if (keyCode === 27)
I would prefer moving to an if...else if statement, or at least updating the switch statement to catch multiple cases for down and enter.